src/snowflake/snowpark/session.py

```diff
@@ -2106,48 +2106,15 @@ def table_function(
         # AST.
         stmt = None
         if _emit_ast:
+            build_intermediate_stmt(self._ast_batch, func_name)
             stmt = self._ast_batch.assign()
-            expr = stmt.expr
-
-            # Session.table_function is a two-in-one:
-            # - Can call functions directly. Use build_session_table_fn_apply
-            # - Can use Callable or TableFunctionCall objects. Use build_table_fn_apply
-            if isinstance(func_name, TableFunctionCall):
-                assert (
-                    func_name._ast is not None
-                ), "TableFunctionCall must have an ast assigned."
-
-                apply_expr = with_src_position(expr.apply_expr, stmt)
-                apply_expr.fn.table_fn.call_type.table_fn_call_type__session_table_fn = (
-                    True
-                )
-                build_table_fn_apply(
-                    expr,
-                    None,  # The callee is a TableFunctionCall object that carries its own name.
-                    func_name._ast,
-                    *func_arguments,
-                    **func_named_arguments,
-                )
-            elif isinstance(func_name, Callable):
-                apply_expr = with_src_position(expr.apply_expr, stmt)
-                apply_expr.fn.table_fn.call_type.table_fn_call_type__session_table_fn = (
-                    True
-                )
-                build_table_fn_apply(
-                    expr,
-                    None,  # The callee is a Callable object that carries its own name.
-                    func_name._ast,  # Inline the AST expression of the callee.
-                    *func_arguments,
-                    **func_named_arguments,
-                )
-            elif isinstance(func_name, str):
-                build_session_table_fn_apply(
-                    expr, func_name, *func_arguments, **func_named_arguments
-                )
-            elif isinstance(func_name, list):
-                build_session_table_fn_apply(
-                    expr, func_name, *func_arguments, **func_named_arguments
-                )
+            ast = with_src_position(stmt.expr.sp_session_table_function, stmt)
```

Comment:
why not `func =  with_src_position(stmt.expr.sp_session_table_function.func, stmt)`, and then pass `func` instead of `ast.func` to build_indirect_table_fn_apply?

Reply:
For consistency with all other occurrences of `ast = with_src_position(stmt.expr.<<sp_specific_variant>>, stmt)` in the Snowpark codebase.